### PR TITLE
fix: Correct *some* problems mkm found with ingester2/querier

### PR DESCRIPTION
**Pull request by carols10cents**
Connects to #6421. Does not completely solve yet.

# Commit 1

The first problem both Dom and I suspected was that the transition shard ID was not matching the hardcoded hack value. The first commit in this PR changes the code to still hardcode the transition shard *index* to 1, but to look up in the catalog what the transition shard *id* ends up being and use that value. This didn't seem to affect the problem in the issue at all as far as I could tell, but was incorrect, so good to change anyway. Ultimately, shards will completely go away when the transition is complete.

# Commit 2

The next problem I noticed is in the test mkm was doing, the steps were:

- Enqueue data
- Query, data is returned
- Ingester persists
- Query, data is not returned

I noticed the querier was not getting any metadata about the ingester UUID and number of files persisted from the second query when it queries the ingester. The ingester was just not returning anything if there were no record batches to return from its memory, but we need to always send back the metadata.

The second commit in this PR changes that, and I'm now able to see the querier get the metadata and the function I expect to say "should I refresh the cache?" is now returning `true` for the second request, but I'm still not seeing the data returned, so there's still another bug somewhere.

I can't write a test for this because ingester2 doesn't implement the write info service yet, so there's not a convenient way for tests to tell that data has been persisted or not.

# Commit 3

The third commit removes the `test_rpc_write` CI job; Dom already made it un-required again. Since the feature flag has been removed, it's exactly the same as the `test` job and is just wasting resources.

### CODE REVIEW

File: ingester2/src/buffer_tree/table.rs [hunk 1]
```diff
@@ -263,38 +267,40 @@ where
...
+                    } else {
+                        Projection::Some(columns.as_ref())
+                    };
+
+                    let data = Box::pin(MemoryStream::new(
+                        data.project_selection(selection).into_iter().collect(),
+                    ));
+                    PartitionResponse::new(data, id, None, completed_persistence_count)
+                }
+                None => PartitionResponse::new_no_batches(id, None, completed_persistence_count),
```
Comment (carols10cents) [inline ingester2/src/buffer_tree/table.rs:299]: Dom: [`datafusion_util::MemoryStream` must have at least one batch](https://github.com/influxdata/influxdb_iox/blob/97d90f5615a1370b43a1548c827dafde5f32c168/datafusion_util/src/lib.rs#L123-L125), so even though creating a `MemoryStream` with an empty vec *compiles*, it panics at runtime.

**Review by pauldix — APPROVED** [review]
Worth having @domodwyer take a look, but I think it's ok for merging.

File: ingester2/src/buffer_tree/table.rs (hunk 1 above)
Comment (domodwyer) [reply ingester2/src/buffer_tree/table.rs:299]: What a nice surprise!

**Comment by domodwyer** [comment]
I took a look over - I think we probably didn't need the ShardId code churn but doesn't hurt - I hope we can remove it soon.

I opened a PR that refactors the constructor stuff to work around the `MemoryStream` stuff and some minor other bits https://github.com/influxdata/influxdb_iox/pull/6436

**Pull request merged**